layer2/lib/test/example_tests.ml

```diff
@@ -0,0 +1,246 @@
+open Tools
+
+let test_empty () = check_lists [] (Mtm.to_list Mtm.empty)
+let test_from_empty_list () = check_lists [] (Mtm.to_list @@ Mtm.from_list [])
+
+let test_from_one_elem_list () =
+  check_lists [ "a", 1 ] (Mtm.to_list @@ Mtm.from_list @@ [ "a", 1 ])
+;;
+
+let test_from_two_elem_list () =
+  check_lists [ "a", 1; "b", 2 ] (Mtm.to_list @@ Mtm.from_list @@ [ "a", 1; "b", 2 ])
+;;
+
+let test_from_three_elem_list () =
+  check_lists
+    [ "a", 1; "b", 2; "c", 3 ]
+    (Mtm.to_list @@ Mtm.from_list @@ [ "b", 2; "a", 1; "c", 3 ])
+;;
+
+let test_remove_from_empty () =
+  check_lists
+    []
+    Mtm.(
+      to_list
+      @@ Tools.get_tree
+      @@ Mtm.execute (Mtm.Remove { key = "non-existent" }) Mtm.empty)
+;;
+
+let test_remove_missing_key () =
+  check_lists
+    ([ "b", 0 ]
+    |> Mtm.from_list
+    |> Mtm.execute (Mtm.Remove { key = "non-existent" })
+    |> Tools.get_tree
+    |> Mtm.to_list)
+    [ "b", 0 ]
+;;
+
+let test_remove_leaf () =
+  check_lists
+    []
+    (Mtm.to_list
+    @@ Tools.get_tree
+    @@ Mtm.execute (Mtm.Remove { key = "a" })
+    @@ Mtm.from_list [ "a", 1 ])
+;;
+
+let test_remove_node_right () =
+  let initial_map = Mtm.from_list [ "a", 1; "b", 2 ] in
+  let _op, _proof, final_map = Mtm.execute (Mtm.Remove { key = "a" }) initial_map in
+  check_lists [ "b", 2 ] (Mtm.to_list final_map)
+;;
+
+let test_remove_node_right_right () =
+  check_lists
+    [ "b", 2; "c", 3 ]
+    (Mtm.to_list
+    @@ Tools.get_tree
+    @@ Mtm.execute (Mtm.Remove { key = "a" })
+    @@ Mtm.from_list [ "a", 1; "b", 2; "c", 3 ])
+;;
+
+let test_remove_node_right_left () =
+  let initial_map = Mtm.from_list [ "a", 1; "c", 3; "b", 2 ] in
+  let _, _, final_map = Mtm.execute (Mtm.Remove { key = "a" }) initial_map in
+  check_lists [ "b", 2; "c", 3 ] (Mtm.to_list final_map)
+;;
+
+let test_remove_2 () =
```

Comment:
how do it get removed?

Reply:
the other comments are miner. but just this one is a puzzle to me. is it expected?